### ISSUE TITLE
Could rewind functionality with `rr` eventually be supported?

### ISSUE THREAD

**Issue by BatmanAoD** [intellij-rust/intellij-rust#1110]
I see that integrating a debugger is an existing issue, #535, and that it's actually working with CLion and nightly Rust (awesome!).

I recently discovered [rr](http://rr-project.org/), a tool for recording program execution and replaying it in `gdb`, *possibly in reverse*. I discovered it from [this blog post](http://huonw.github.io/blog/2015/10/rreverse-debugging/), in which the author shows how to use `rust-gdb` with `rr`.

`rr` is unfortunately only available on Linux, but for platforms that support it, would it be possible to eventually use it with the intellij-rust debugger integration?

**Comment by matklad** [comment]
Yeah, having some undo debugging would be great! There's some for CLion already (not rr based): https://blog.jetbrains.com/clion/2016/09/undo-for-clion/.

That said, it's a rather far future, we have a **ton** of code to write until the basic debugging works properly :)
